lib/Analysis/Allocation.cpp

```diff
@@ -84,8 +84,8 @@ ScratchConfig getScratchConfigForCvt(RankedTensorType srcTy,
 
   assert(cvtNeedsSharedMemory(srcTy, dstTy));
 
-  auto inOrd = gpu::getOrder(srcLayout);
-  auto outOrd = gpu::getOrder(dstLayout);
+  const auto &inOrd = gpu::getOrder(srcLayout);
+  const auto &outOrd = gpu::getOrder(dstLayout);
```

Comment:
Can you take this result by reference? Is it one of those cases where the lifetime is extended?

At any rate, a better fix would be to make `getOrder` have NRVO semantics

Reply:
>  Is it one of those cases where the lifetime is extended?

Yes, I would also like to point out that this was suggested by https://scan.coverity.com/ tool, which we use in XPU backend mostly for our files, but since some of it is duplicated, I thought it would be nice to push even such minor changes upstream to improve the code a bit.

> At any rate, a better fix would be to make getOrder have NRVO semantics, but it's pretty much a microoptimisation at this point

Unfortunately I'm not sure what needs to be done for this. Can we leave it as is until this code is considered part of the hottest path?